src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return this[0].toString();
+    }
```

Comment:
Better remove this line. Because for case with `this.length === 1` without this code it wouldn't be increasing algorithmic complexity. Because you cycle would have only one iteration, and creating variables - not so long process. But this way your code would be shorter.

Reply:
thnks, great idea, completely agree with u